src/SignalR/common/Http.Connections/src/ConnectionEndpointRouteBuilderExtensions.cs

```diff
@@ -0,0 +1,144 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Connections;
+using Microsoft.AspNetCore.Http.Connections;
+using Microsoft.AspNetCore.Http.Connections.Internal;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Microsoft.AspNetCore.Routing
+{
+    public static class ConnectionEndpointRouteBuilderExtensions
+    {
+        /// <summary>
+        /// Maps incoming requests with the specified path to the provided connection pipeline.
+        /// </summary>
+        /// <param name="builder">The <see cref="IEndpointRouteBuilder"/></param>
+        /// <param name="pattern">The request path.</param>
+        /// <param name="configure">A callback to configure the connection.</param>
+        public static IEndpointConventionBuilder MapConnections(this IEndpointRouteBuilder builder, string pattern, Action<IConnectionBuilder> configure) =>
```

Comment:
Why are the extension methods pattern?

Reply:
MVC is using template in `MapControllerRoute` because it's consistent with a lot of cases where people had `MapRoute(name: "default", template: "...")`  - I'd prefer pattern for new stuff. I'd be willing to consider the naming in MVC if it really bugs yall